tests/testthat/test-utils-ui.R

```diff
@@ -58,3 +58,94 @@ test_that("bulletize() works", {
   expect_snapshot(cli::cli_bullets(bulletize(letters[1:6], n_fudge = 0)))
   expect_snapshot(cli::cli_bullets(bulletize(letters[1:8], n_fudge = 3)))
 })
+
+# menu(), but based on readline() + cli and mockable ---------------------------
```

Comment:
This basically covers the `cli_menu()` stuff 100%, except the lines you can only hit in interactive use.